### ISSUE TITLE
No way to stop text content wrapping floated content

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#1473]
Gutenberg has a Button block which is great, as it will allow people to easily add buttons into their content without having to use shortcodes. One issue with this though, is if you left or right align the button and then try to add a text block after it, the text wraps around the button. This means that you can't have a button on it's own line. You end up with something like this...

![gutenberg-clearingblocks](https://user-images.githubusercontent.com/2078245/27569794-d18c6f3e-5b3c-11e7-8a6f-bf599d2c22f1.png)

People are going to want to be able to insert buttons without having the following paragraph content wrap around the button. There needs to be a way to set a block to automatically clear so that you can end up with something like...

![gutenberg-clearingblocks2](https://user-images.githubusercontent.com/2078245/27569905-9031b412-5b3d-11e7-8dd0-e1d91623f210.png)

The only way I could do the above is by adding a adding a blank Text block and inserting half a dozen empty lines into it, which is extremely messy.

**Comment by truongwp** [comment]
Same with the image. I think should have a style option for image, button... to choose between Block and Inline.

**Comment by maddisondesigns** [comment]
@truongwp Agreed. That sounds like a good solution 👍🏼

**Comment by truongwp** [comment]
I created a pull request for adding that option for Image block because I meet another error with Button block.

**Comment by mtias** [comment]
We can add a clear toggle on the button inspector. @jasmussen thoughts?

**Comment by jasmussen** [comment]
> We can add a clear toggle on the button inspector. @jasmussen thoughts?

Yep, I think so. Need a good label for it. I can't think of any. But yes, it feels like an advanced layout task, and so the inspector is good for it.

**Comment by truongwp** [comment]
"Start this block on a new line" is good but too long. I will try and hope that the UI is still ok.

**Comment by jwold** [comment]
> "Start this block on a new line" is good but too long. I will try and hope that the UI is still ok.

What if we tried "force new line"?

**Comment by truongwp** [comment]
I used the text "Start on a new line". But I think there will be some discussions for this problem.

**Comment by jasmussen** [comment]
Another option is a "Clear" block ;)

**Comment by jwold** [comment]
> Another option is a "Clear" block ;)

Pretty sure that was in jest, but in reality that would solve the problem :D

**Comment by jasmussen** [comment]
Oh sorry I see you already suggested this. I think Matías also suggested this in a DM with me ;)

The point is, to me this feels like a somewhat advanced layouting issue. Which is cool, I like those. But if you're doing something this advanced, then perhaps you either already know what "clearing floats" means, or you might be able to learn it from tinkering. 

The key is finding a balance here that doesn't overload the interface. A clear block would mean we didn't need a "clear preceeding blocks" toggle in the inspector for all blocks.

**Comment by jwold** [comment]
> The point is, to me this feels like a somewhat advanced layouting issue. Which is cool, I like those. But if you're doing something this advanced, then perhaps you either already know what "clearing floats" means, or you might be able to learn it from tinkering.

Great thoughts. If someone understands they have this problem, tinkering around to solve it wouldn't be the worst thing. I think you're suggesting that either having the "clear" block in the inserter or adding the setting to post settings could work?

**Comment by paaljoachim** [comment]
Relooking at this issue:

I see no way to wrap around an image today. Version 0.6. 

![screen shot 2017-08-04 at 01 52 06](https://user-images.githubusercontent.com/5323259/28948816-97e171be-78b8-11e7-98fe-7b06aeceb7db.png)


There is mention of adding some wrap controls it seems to the settings area. As it would be good to have a toogle in the Block that says wrap. Turn it on or turn it off.

**Comment by mtias** [comment]
@paaljoachim this works for me in master.

![image](https://user-images.githubusercontent.com/548849/28970021-3b71ac72-7927-11e7-9ff3-b1cc5f77483a.png)

**Comment by paaljoachim** [comment]
I got two things going here. The two images are aligned either left or right have no move up or down arrows beside them. Clicking center icon the up/down arrows show up again.

Clicking down arrow in a paragraph filled with text. It then moves down and the image is wrapped. 

Here is a short example of what happened when I did this:

![gutenberg-images](https://user-images.githubusercontent.com/5323259/28970765-0e545afc-792a-11e7-864f-2ac501a68d25.gif)

At the moment it is kinda messy.
